src/functionalTest/java/uk/gov/hmcts/reform/pip/data/management/publication/PublicationTest.java

```diff
@@ -0,0 +1,201 @@
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.springframework.http.HttpStatus.CREATED;
+import static org.springframework.http.HttpStatus.OK;
+import static uk.gov.hmcts.reform.pip.data.management.utils.TestUtil.BEARER;
+import static uk.gov.hmcts.reform.pip.data.management.utils.TestUtil.randomLocationId;
+
+@ExtendWith(SpringExtension.class)
+@ActiveProfiles(profiles = "functional")
+@SuppressWarnings({"PMD.NcssCount"})
+@AutoConfigureEmbeddedDatabase(type = AutoConfigureEmbeddedDatabase.DatabaseType.POSTGRES)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@SpringBootTest(classes = {Application.class, OAuthClient.class},
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class PublicationTest extends FunctionalTestBase {
+    @Value("${test-user-id}")
+    private String userId;
+
+    private static final String PUBLICATION_URL = "/publication";
+    private static final String ARTEFACT_BY_LOCATION_ID_URL = PUBLICATION_URL + "/locationId/";
+    private static final String ARTEFACT_BY_SEARCH_VALUE_URL = PUBLICATION_URL + "/search/";
+
+    private static final String TESTING_SUPPORT_LOCATION_URL = "/testing-support/location/";
+    private static final String TESTING_SUPPORT_PUBLICATION_URL = "/testing-support/publication/";
+    private static final ArtefactType ARTEFACT_TYPE = ArtefactType.LIST;
+    private static final Sensitivity SENSITIVITY = Sensitivity.PUBLIC;
+    private static final String PROVENANCE = "MANUAL_UPLOAD";
+    private static final String SOURCE_ARTEFACT_ID = "sourceArtefactId";
+    private static final LocalDateTime DISPLAY_FROM = LocalDateTime.now().truncatedTo(ChronoUnit.SECONDS);
+    private static final Language LANGUAGE = Language.ENGLISH;
+    private static final ListType LIST_TYPE = ListType.CIVIL_DAILY_CAUSE_LIST;
+    private static final String COURT_ID = randomLocationId();
+    private static final String USER_ID_HEADER = "x-user-id";
+    private static final String ISSUER_HEADER = "x-issuer-id";
+    private static final String COURT_NAME = "TestLocation" + COURT_ID;
+    private static final String CASE_NUMBER = "4568454842";
+    private static final String EMAIL = "test@hmcts.net";
+
+    private static final LocalDateTime CONTENT_DATE = LocalDateTime.now().toLocalDate().atStartOfDay()
+        .truncatedTo(ChronoUnit.SECONDS);
+
+    @BeforeAll
+    public void setup() {
+        doPostRequest(
+            TESTING_SUPPORT_LOCATION_URL + COURT_ID,
+            Map.of(AUTHORIZATION, BEARER + accessToken), COURT_NAME
+        );
+    }
+
+    @AfterAll
+    public void teardown() {
+        doDeleteRequest(TESTING_SUPPORT_PUBLICATION_URL + COURT_NAME, Map.of(AUTHORIZATION, BEARER + accessToken));
+        doDeleteRequest(TESTING_SUPPORT_LOCATION_URL + COURT_NAME, Map.of(AUTHORIZATION, BEARER + accessToken));
+    }
+
+    @Test
+    void shouldAbleToUploadThePublication() throws Exception {
```

Comment:
The asserts and checks look good, however I think I'd split this out a bit rather than then being in a single test. I'd suggest grouping this by endpoint instead:

1) POST JSON Publication
2) POST Flat File Publication
2) Get Metadata for Publication
3) Get Payload for Publication
4) Get File for Publication
5) Delete Publication
6) Get artefacts by Search Value
7) Get artefacts by Location

Reply:
Done:)